controllers/clusterpullsecret_controller.go

```diff
@@ -42,10 +42,26 @@ func (r *ClusterPullSecretReconciler) Reconcile(req ctrl.Request) (ctrl.Result,
 	var pullSecret v1.ClusterPullSecret
 	if err := r.Get(ctx, req.NamespacedName, &pullSecret); err != nil {
 		r.Log.Info(fmt.Sprintf("%s\n", errors.Wrap(err, "unable to fetch pullSecret")))
+
+		serviceaccounts := &corev1.ServiceAccountList{}
```

Comment:
I'd want to see this separated into new methods / functions that have clear names, so that the flow can be followed. Otherwise this code becomes rather confusing to read and maintain. Does that sound fair?